impeller/compiler/shader_lib/impeller/path.glsl

```diff
@@ -121,7 +121,7 @@ QuadDecomposition DecomposeQuad(QuadData quad, float tolerance) {
   // This should never happen, but if it does happen be more defensive -
   // otherwise we'll get NaNs down the line.
   if (dd == vec2(0.)) {
-    return QuadDecomposition(0., 0., 0., 0., 0, 0.);
+    dd = vec2(1.0, 1.0);
```

Comment:
This is required to render correctly compared to the current tessellator

Reply:
This is the case where the quad is a straight line right?

In that case, the CPU based code just unconditionally puts p2 into the line.

I think here it would work the same if we did `return QuadDecomposition(0., 0., 0., 0., 1., 0.);` - to indicate that this generates a single line (right now it's indicating that it generates no lines).

Reply:
Yeah, this works!